src/backend/gporca/libgpopt/src/xforms/CXformUpdate2DML.cpp

```diff
@@ -141,30 +151,47 @@ CXformUpdate2DML::Transform(CXformContext *pxfctxt, CXformResult *pxfres,
 		pexprAssertConstraints = pexprSplit;
 	}
 
-	const ULONG num_cols = pdrgpcrInsert->Size();
-
-	CBitSet *pbsModified = GPOS_NEW(mp) CBitSet(mp, ptabdesc->ColumnCount());
-	for (ULONG ul = 0; ul < num_cols; ul++)
+	CExpression *pexprDML = NULL;
+	// create logical DML
+	ptabdesc->AddRef();
+	if (fSplit)
 	{
-		CColRef *pcrInsert = (*pdrgpcrInsert)[ul];
-		CColRef *pcrDelete = (*pdrgpcrDelete)[ul];
-		if (pcrInsert != pcrDelete)
+		const ULONG num_cols = pdrgpcrInsert->Size();
```

Comment:
Why was this line moved here? In original commit it was not touched.

Reply:
I thought it made sense since its value is only used inside if.

Reply:
Yes, but it is backport

Reply:
I think we should preserve the original commit code as much as possible. To make it easier to backport other patches that depend on this code.